File: example/google.ts
import { Builder, Browser } from 'selenium-webdriver';
// import { ServiceBuilder } from 'selenium-webdriver/firefox';

import { AnthropicPlanner, BrowserAgent, pauseForInput } from 'cerebellum-ai';


(async function example() {
  let driver = await new Builder()
    .forBrowser(Browser.CHROME)
    // .setFirefoxService(new ServiceBuilder('/snap/bin/geckodriver')) // Necessary for snap based firefox installs
    .build();

  try {
    await driver.get('https://www.google.com');
    
    // Define your goal
    const goal = 'Show me the wikipedia page of the creator of Bitcoin';
    
    // Create the Cerebellum browser agent
    const planner = new AnthropicPlanner(process.env.ANTHROPIC_API_KEY as string);
    const agent = new BrowserAgent(driver, planner, goal, {
      pauseAfterEachAction: true,
    });

    // Have Cerebellum takeover website navigation
    await agent.start();

    // Goal has now been reached, you may interact with the Selenium driver any way you want
    await pauseForInput();
    
  } finally {
    await driver.quit();
  }
})();

File: python/examples/google.py
from seleniumbase import get_driver
from cerebellum import vLLMPlanner, vLLMPlannerOptions, BrowserAgent, BrowserAgentOptions, pause_for_input

def main():
    driver = get_driver()

    try:
        # Set your starting page
        driver.get("https://www.google.com")

        # Define your goal
        goal = "Show me the wikipedia page of the creator of Bitcoin"

        options = vLLMPlannerOptions(
            model="Qwen2-VL-7B-Instruct",
            server="http://localhost:8000",
            debug_image_path='debug.png'
        )

        # Create the Cerebellum browser agent
        planner = vLLMPlanner(options)

        options = BrowserAgentOptions(pause_after_each_action=True)

        agent = BrowserAgent(driver, planner, goal, options)

        # pause_for_input()
        # Have Cerebellum takeover website navigation
        agent.start()

        # Goal has now been reached, you may interact with the Selenium driver any way you want
        pause_for_input()

    finally:
        driver.quit()


if __name__ == "__main__":
    main()
